api/lib/opentelemetry/instrumentation/adapter.rb

```diff
@@ -0,0 +1,244 @@
+    #           # a monkey patch
+    #         end
+    #
+    #         # determine if the target library is present
+    #         present do
+    #           Gem.loaded_specs.key?('sinatra')
+    #         end
+    #
+    #         # if the target library is present, is it compatible?
+    #         compatible do
+    #           Gem.loaded_specs['sinatra'].version > MIN_VERSION
+    #         end
+    #       end
+    #     end
+    #   end
+    # end
+    #
+    # The adapter name and version will inferred from the namespace of the
+    # class. In this example, they'd be 'OpenTelemetry::Adapters::Sinatra' and
+    # OpenTelemetry::Adapters::Sinatra::VERSION, but can be explicitly set using
+    # the +adapter_name+ and +adapter_version+ methods if necessary.
+    #
+    # All subclasses of OpenTelemetry::Instrumentation::Adapter are automatically
+    # registered with OpenTelemetry.instrumentation_registry which is used by
+    # SDKs for for instrumentation discovery and installation.
+    #
+    # Instrumentation libraries can use the adapter subclass to easily gain
+    # a reference to its named tracer. For example:
+    #
+    # OpenTelemetry::Adapters::Sinatra.instance.tracer
+    #
+    # The adapter class establishes a convention for disabling an adapter
+    # by environment variable and local configuration. An adapter disabled
+    # by environment variable will take precedence over local config. The
+    # convention for environment variable name is the library name, upcased with
+    # '::' replaced by underscores, and '_ENABLED' appended. For example:
+    # OPENTELEMETRY_ADAPTERS_SINATRA_ENABLED = false.
+    class Adapter
+      class << self
+        NAME_REGEX = /^(?:(?<namespace>[a-zA-Z0-9_:]+):{2})?(?<classname>[a-zA-Z0-9_]+)$/.freeze
+        private_constant :NAME_REGEX
+
+        private :new # rubocop:disable Style/AccessModifierDeclarations
+
+        def inherited(subclass)
+          OpenTelemetry.instrumentation_registry.register(subclass)
+        end
+
+        # Optionally set the name of this instrumentation adapter. If not
+        # explicitly set, the name will default to the namespace of the class,
+        # or the class name if it does not have a namespace. If there is not
+        # a namespace, or a class name, it will default to 'unknown'.
+        #
+        # @param [String] adapter_name The full name of the adapter package
+        def adapter_name(adapter_name = nil)
+          if adapter_name
+            @adapter_name = adapter_name
+          else
+            @adapter_name ||= infer_name || 'unknown'
+          end
+        end
+
+        # Optionally set the version of this adapter. If not explicitly set,
+        # the version will default to the version constant under namespace of
+        # the class, or the version constant under the class name if it does not
+        # have a namespace. If a version constant cannot be found, it defaults
```

Comment:
More accurate to refer to `VERSION constant` instead of `version constant` in this comment?

Reply:
Agreed, I think `VERSION constant` expresses what you're doing.